docs/doc_example/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: Apache-2.0
+
+add_custom_target(doc-example)
+set_target_properties(doc-example PROPERTIES FOLDER "Docs")
+
+add_custom_target(check-doc-example
+  COMMENT "Running the documentation example"
+  COMMAND "${CMAKE_CTEST_COMMAND}" -L doc-example --output-on-failure -C $<CONFIG> --force-new-ctest-process
+  USES_TERMINAL
+  DEPENDS doc-example
+  )
+
+set_target_properties(check-doc-example PROPERTIES FOLDER "Docs")
+# Exclude the target from the default VS build
+set_target_properties(check-doc-example PROPERTIES EXCLUDE_FROM_DEFAULT_BUILD ON)
+
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/gen_add_onnx.py
+  ${CMAKE_CURRENT_BINARY_DIR}/gen_add_onnx.py
+  COPYONLY
+  )
+
+if (WIN32)
+  # On Windows, we need a .def file to specify the export functions from a DLL.
+  # The name of the .def file must match the name of the DLL being created
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/add.def
+    ${CMAKE_CURRENT_BINARY_DIR}/add.def
+    COPYONLY
+    )
+endif()
+
+set(OMRuntimeTestModelLibraryFiles ${CMAKE_CURRENT_BINARY_DIR}/add${CMAKE_SHARED_LIBRARY_SUFFIX})
+if (WIN32)
+  list(APPEND OMRuntimeTestModelLibraryFiles ${CMAKE_CURRENT_BINARY_DIR}/add${CMAKE_LINK_LIBRARY_SUFFIX})
+endif()
+
+add_custom_command(
+  OUTPUT ${OMRuntimeTestModelLibraryFiles}
+  COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/gen_add_onnx.py
+  COMMAND onnx-mlir add.onnx
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  DEPENDS onnx-mlir
+  )
+
+add_custom_target(OMRuntimeTestModel
+  DEPENDS ${OMRuntimeTestModelLibraryFiles}
+  )
+
+add_library(OMRuntimeTestModelLibrary SHARED IMPORTED)
+add_dependencies(OMRuntimeTestModelLibrary OMRuntimeTestModel)
+set_target_properties(OMRuntimeTestModelLibrary PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/add${CMAKE_SHARED_LIBRARY_SUFFIX}
+  )
+if (WIN32)
+  set_target_properties(OMRuntimeTestModelLibrary PROPERTIES
+    IMPORTED_IMPLIB ${CMAKE_CURRENT_BINARY_DIR}/add${CMAKE_LINK_LIBRARY_SUFFIX}
+    )
+endif()
+
+add_onnx_mlir_executable(OMRuntimeTest EXCLUDE_FROM_ALL
+  main.c
+
+  NO_INSTALL
+
+  INCLUDE_DIRS PRIVATE
+  ${ONNX_MLIR_SRC_ROOT}/include
+
+  LINK_LIBS PRIVATE
+  OMRuntimeTestModelLibrary
+  cruntime
+  )
+set_output_directory(OMRuntimeTest BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} LIBRARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+add_test(NAME OMRuntimeTest COMMAND OMRuntimeTest WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+set_property(TARGET OMRuntimeTest PROPERTY FOLDER "Docs")
+set_tests_properties(OMRuntimeTest PROPERTIES LABELS doc-example)
+
```

Comment:
The expectation is that `make test` should not fail after `make`. So yes as long as we still have `make test`, either we should make `OMRuntimeTest` work with `make test`, or we should remove it from `make test`.

Reply:
@sstamenova @gongsu832 Does something need to be done here?

Reply:
I made the change to make `make test` continue to work as before by adding the `ONNX_MLIR_BUILD_TESTS` option. This allows `make test` to stay as it is unless `ONNX_MLIR_BUILD_TESTS` is set to `OFF` at configuration time.